Nio/Conversations/RoomView.swift

```diff
@@ -41,7 +42,32 @@ struct RoomContainerView: View {
                 self.eventToReactTo = nil
             }
         }
-        .onAppear { self.room.markAllAsRead() }
+        .alert(isPresented: $showJoinAlert) {
+            let roomName = self.room.summary.displayname ?? self.room.summary.roomId ?? "New Conversation"
+            return Alert(
+                title: Text("Join Conversation?"),
+                message: Text("Accept invitation to join '\(roomName)'?"),
+                primaryButton: .default(
+                    Text("Join"),
+                    action: {
+                        self.room.room.mxSession.joinRoom(self.room.room.roomId) { _ in
+                            self.room.markAllAsRead()
+                        }
+                    }),
+                secondaryButton: .cancel())
+        }
+        .onAppear {
+            switch(self.room.summary.membership) {
```

Comment:
No need for the parens around the conditional here.
```suggestion
            switch self.room.summary.membership {
```